jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/providers/ScanningAppProvider.java

```diff
@@ -149,62 +134,18 @@ protected App createApp(Path path)
         App app = new App(_deploymentManager, this, path);
         if (LOG.isDebugEnabled())
             LOG.debug("{} creating {}", this, app);
-
-        String defaultEnvironmentName = _deploymentManager.getDefaultEnvironmentName();
-
-        String environmentName = app.getEnvironmentName();
-        if (StringUtil.isBlank(environmentName) && StringUtil.isNotBlank(defaultEnvironmentName))
-        {
-            environmentName = defaultEnvironmentName;
-            app.getProperties().put(Deployable.ENVIRONMENT, environmentName);
-            if (LOG.isDebugEnabled())
-                LOG.debug("{} default environment for {}", this, app);
-        }
-
-        if (StringUtil.isNotBlank(environmentName))
-        {
-            // If the app specifies the environment for this provider, then this deployer will deploy it.
-            if (environmentName.equalsIgnoreCase(getEnvironmentName()))
-            {
-                if (LOG.isDebugEnabled())
-                    LOG.debug("{} created {}", this, app);
-                return app;
-            }
-
-            // If we are the default provider then we may warn
-            if (getEnvironmentName().equalsIgnoreCase(defaultEnvironmentName))
-            {
-                // if the app specified an environment name, then produce warning if there is no provider for it.
-                if (!_deploymentManager.hasAppProviderFor(environmentName))
-                    LOG.warn("No AppProvider with environment {} for {}", environmentName, app);
-                return null;
-            }
-        }
-
-        LOG.warn("{} no environment for {}, ignoring", this, app);
-        return null;
+        return app;
```

Comment:
Maybe fix the comment to this method. The path can be the path to the `.war` or war directory also, no?